test/PackUnPackTest.cc

```diff
@@ -0,0 +1,678 @@
+                         area_height_,
+                         in_compn_bit_buffer1_,
+                         in_compn_bit_buffer2_);
+
+            EXPECT_FALSE(HasFailure());
+        }
+    }
+
+    uint8_t *local_cache1_, *local_cache2_;
+    uint8_t *in_bit_buffer_;
+    uint32_t in_stride_, out_stride_;
+    uint8_t *in_compn_bit_buffer1_, *in_compn_bit_buffer2_;
+    uint32_t area_width_, area_height_;
+    uint32_t test_size_;
+};
+
+TEST_P(PackTest, PackTest) {
+    run_test();
+};
+
+INSTANTIATE_TEST_CASE_P(PACK, PackTest, ::testing::ValuesIn(TEST_PACK_SIZES));
+
+class Pack2dTest : public ::testing::Test,
+                   public ::testing::WithParamInterface<AreaSize> {
+  public:
+    Pack2dTest()
+        : area_width_(std::get<0>(GetParam())),
+          area_height_(std::get<1>(GetParam())) {
+        in_stride_ = out_stride_ = MAX_SB_SIZE;
+        test_size_ = MAX_SB_SQUARE;
+    }
+
+    void SetUp() override {
+        in_bit_buffer_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        in_8bit_buffer_ =
+            reinterpret_cast<uint8_t *>(eb_aom_memalign(32, test_size_));
+        out16_bit_buffer1_ =
+            reinterpret_cast<uint16_t *>(eb_aom_memalign(32, 2 * test_size_));
+        out16_bit_buffer2_ =
+            reinterpret_cast<uint16_t *>(eb_aom_memalign(32, 2 * test_size_));
+        out16_bit_buffer3_ =
+            reinterpret_cast<uint16_t *>(eb_aom_memalign(32, 2 * test_size_));
+        memset(
+            out16_bit_buffer1_, 0, test_size_ * sizeof(out16_bit_buffer1_[0]));
+        memset(
+            out16_bit_buffer2_, 0, test_size_ * sizeof(out16_bit_buffer2_[0]));
+        memset(
+            out16_bit_buffer3_, 0, test_size_ * sizeof(out16_bit_buffer3_[0]));
+    }
+
+    void TearDown() override {
+        if (in_bit_buffer_)
+            eb_aom_free(in_bit_buffer_);
+        if (in_8bit_buffer_)
+            eb_aom_free(in_8bit_buffer_);
+        if (out16_bit_buffer1_)
+            eb_aom_free(out16_bit_buffer1_);
+        if (out16_bit_buffer2_)
+            eb_aom_free(out16_bit_buffer2_);
+        if (out16_bit_buffer3_)
+            eb_aom_free(out16_bit_buffer3_);
+    }
+
+  protected:
+    void check_output(uint32_t width, uint32_t height, uint16_t *out_1,
+                      uint16_t *out_2) {
+        int fail_count = 0;
+        for (uint16_t j = 0; j < height; j++) {
+            for (uint16_t k = 0; k < width; k++) {
+                if (out_1[k + j * out_stride_] != out_2[k + j * out_stride_])
+                    fail_count++;
+            }
+        }
+        EXPECT_EQ(0, fail_count)
+            << "compare result error"
+            << "in test area for " << fail_count << "times";
+    }
+
+    void run_2d_test() {
+        for (int i = 0; i < 3; i++) {
+            eb_buf_random_u8(in_bit_buffer_, test_size_);
+            eb_buf_random_u8(in_8bit_buffer_, test_size_);
+
+            eb_enc_msb_pack2d_avx2_intrin_al(in_bit_buffer_,
```

Comment:
The first parameter of the function is in8_bit_buffer, why use in_bit_buffer? Meantime the in_bit_buffer could be renamed to inn_bit_buffer to match the function declaration.